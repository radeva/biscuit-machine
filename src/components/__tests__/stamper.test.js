import React from 'react';
import Stamper from '../stamper';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

describe('Stamper', () => {
  it('should be rendered correctly when machine is off', () => {
    const { queryByTestId, container } = render(
      <Stamper
        hasBiscuitToStamp={false}
        isMachineMovementOn={false}
        isMachineMovementPaused={false}
      />,
    );

    expect(container).toMatchSnapshot();
    expect(queryByTestId('raw-biscuit')).not.toBeInTheDocument();
  });

  it('should be rendered correctly when machine is on', () => {
    const { getByTestId } = render(
      <Stamper
        hasBiscuitToStamp={false}
        isMachineMovementOn={true}
        isMachineMovementPaused={false}
      />,
    );

    expect(getByTestId('stamper-pusher')).toHaveClass('stamper-pusher-on');
  });

  it('should be rendered correctly when machine is paused', () => {
    const { getByTestId } = render(
      <Stamper
        hasBiscuitToStamp={false}
        isMachineMovementOn={false}
        isMachineMovementPaused={true}
      />,
    );

    expect(getByTestId('stamper-pusher')).toHaveClass('stamper-pusher-on');
    expect(getByTestId('stamper-pusher')).toHaveClass('animation-paused');
  });
});

describe('Biscuit dough', () => {
  it('should be rendered correctly when machine is on', () => {
    const { queryByTestId } = render(
      <Stamper
        hasBiscuitToStamp={true}
        isMachineMovementOn={true}
        isMachineMovementPaused={false}
      />,
    );

    expect(queryByTestId('raw-biscuit')).toBeInTheDocument();
  });

  it('should have animation stopped when machine is paused', () => {
    const { queryByTestId } = render(
      <Stamper
        hasBiscuitToStamp={true}
        isMachineMovementOn={false}
        isMachineMovementPaused={true}
      />,
    );

    expect(queryByTestId('raw-biscuit')).toBeInTheDocument();
    expect(queryByTestId('raw-biscuit')).toHaveClass('animation-paused');
  });
});
