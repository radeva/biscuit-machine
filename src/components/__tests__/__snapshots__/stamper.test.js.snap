// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stamper should be rendered correctly when machine is off 1`] = `
<div>
  <div
    class="stamper left"
  >
    <svg>
      stamper.svg
    </svg>
    <svg
      class="stamper-pusher "
      data-testid="stamper-pusher"
    >
      stamper-pusher.svg
    </svg>
    
  </div>
</div>
`;
